### PR TITLE
pin opam packages for stockholm

### PR DESCRIPTION
**Pull request by lippirk**

### CODE REVIEW

File: tools/xs-opam-ci.env [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export OCAML_VERSION="4.08"
 export DISTRO="debian-unstable"
 export BASE_REMOTE="https://github.com/xapi-project/xs-opam.git"
+export BASE_REMOTE_BRANCH="release/stockholm/lcm"
```
Comment (lindig) [inline tools/xs-opam-ci.env:4]: Why do we need this? This refers to the Jenkins build system and I don't understand the relevance here.

Comment (psafont) [reply tools/xs-opam-ci.env:4]: This is needed to test other repositories against this branch when using travis.

See https://github.com/xapi-project/xen-api/pull/4132

File: packages/xs-extra-dummy/xs-toolstack.master/opam [hunk 2]
```diff
@@ -54,7 +54,7 @@ depends: [
   "xen-api-client"
   "xen-api-client-async"
   "xen-api-sdk"
-  "xenctrl"
+  "xenctrl" {= "0.1.0"}
```
Comment (psafont) [inline packages/xs-extra-dummy/xs-toolstack.master/opam:57]: Is this breaking the tests?

```
#=== ERROR while compiling xenctrl.dummy ======================================#

# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.08.1 | file:///mnt

# path                 ~/.opam/4.08/.opam-switch/build/xenctrl.dummy

# command              /usr/bin/rsync -avz /usr/lib64/ocaml/xenctrl/ /home/opam/.opam/4.08/lib/xenctrl/

# exit-code            23

# env-file             ~/.opam/log/xenctrl-961-0c8b92.env

# output-file          ~/.opam/log/xenctrl-961-0c8b92.out

### output ###

# sending incremental file list

# rsync: change_dir "/usr/lib64/ocaml/xenctrl" failed: No such file or directory (2)

# created directory /home/opam/.opam/4.08/lib/xenctrl

# 

# sent 20 bytes  received 68 bytes  176.00 bytes/sec

# total size is 0  speedup is 0.00

# rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1207) [sender=3.1.3]
```

Comment (lindig) [reply packages/xs-extra-dummy/xs-toolstack.master/opam:57]: I suspect so - why do we need the version constraint?

File: tools/xs-opam-ci.env (hunk 1 above)
Comment (lindig) [reply tools/xs-opam-ci.env:4]: I didn't pay attention to the target branch

File: packages/xs-extra-dummy/xs-toolstack.master/opam (hunk 2 above)
Comment (lippirk) [reply packages/xs-extra-dummy/xs-toolstack.master/opam:57]: this is confusing; i had this failure on my machine, and adding that constraint fixed it- i'll investigate

Comment (lippirk) [reply packages/xs-extra-dummy/xs-toolstack.master/opam:57]: i've versioned `xenctrl.dummy` as version 0 and the real xenctrl as v 1, so hopefully opam will pick the correct one now

File: packages/xs-extra/xenctrl.0.1.0/opam [hunk 3]
```diff
@@ -24,5 +24,5 @@ depexts: [
 
 dev-repo: "git+https://github.com/xapi-project/xenctrl.git"
 url {
-  src: "https://github.com/xapi-project/xenctrl/archive/master.tar.gz"
+  src: "https://github.com/xapi-project/xenctrl/archive/v0.1.0.tar.gz"
```
Comment (psafont) [inline packages/xs-extra/xenctrl.0.1.0/opam:27]: The commit has two tags: v1.1.0 and v0.1.0, which is confusing

File: packages/xs-extra/rrddump.1.33.0/opam [hunk 4]
```diff
@@ -10,5 +10,5 @@ depends: ["xapi-rrd" "xmlm"]
 build: ["dune" "build" "-p" name "-j" jobs]
 dev-repo: "git+https://github.com/xapi-project/xcp-rrdd.git"
 url {
-  src: "https://github.com/xapi-project/xcp-rrdd/archive/master.tar.gz"
+  src: "https://github.com/xapi-project/xcp-rrdd/archive/v1.33.0.tar.gz"
```
[the remark this answers is not in the document]
Comment (lippirk) [reply packages/xs-extra/rrddump.1.33.0/opam:13]: no, we moved rrddump to xcp-rrdd

we should archive https://github.com/xapi-project/rrddump

File: packages/xs-extra/xenctrl.0.1.0/opam (hunk 3 above)
Comment (lippirk) [reply packages/xs-extra/xenctrl.0.1.0/opam]: this has been remedied

**Pull request merged**